src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -409,20 +455,37 @@ void OTARequestor::OnConnectionFailure(void * context, PeerId peerId, CHIP_ERROR
     default:
         break;
     }
+
+    // Give driver a chance to schedule another query
+    requestorCore->mOtaRequestorDriver->UpdateNotFound(UpdateNotFoundReason::ConnectionFailed, chip::System::Clock::Seconds32(0));
```

Comment:
Will this just forever retry until the connection succeeds?

Reply:
It is up to the driver implementation to decide what to do here. In the current implementation there will be a retry if there is a different provider, otherwise -- no.

Reply:
Created https://github.com/project-chip/connectedhomeip/issues/15691 to track this

Reply:
`HandleError` schedules the periodic timer for the default provider -- it has a long timeout (24 hrs is the default) and it is a continuous thing for as long as the device is running. Here we've encountered an error, set our state to `kIdle` and restarted the periodic timer -- this is standard logic for any error. 

Now with this particular error (connection failure) the driver might want to try a different provider right away -- that's what the call to `UpdateNotFound` will achieve. If there is a different provider (and this logic can be extended and refined to include retry count etc) the driver will trigger a query to it immediately.